bookkeeper-server/src/main/java/org/apache/bookkeeper/client/EnsemblePlacementPolicy.java

```diff
@@ -50,16 +55,21 @@
      * @param featureProvider feature provider
      * @param statsLogger stats logger
      */
-    public EnsemblePlacementPolicy initialize(ClientConfiguration conf,
+    public default EnsemblePlacementPolicy initialize(ClientConfiguration conf,
                                               Optional<DNSToSwitchMapping> optionalDnsResolver,
                                               HashedWheelTimer hashedWheelTimer,
                                               FeatureProvider featureProvider,
-                                              StatsLogger statsLogger);
+                                              StatsLogger statsLogger) {
+        return initialize(conf);
```

Comment:
the way I suggested doesn't prevent you calling initialize(conf), no? 

- if your placement policy implements the interface directly, you anyway already implement initialize(conf)
- if you placement policy extends the default ensemble placement policy, this default ensemble placement policy implements the new initialize method hence the old initialize method is already implemented.

Your approach looks messy because the dependency is somehow very strange. if you follow the rule I suggested, it is very clear.